paimon-common/src/main/java/org/apache/paimon/CoreOptions.java

```diff
@@ -1692,20 +1692,28 @@ public boolean deletionVectorsEnabled() {
 
     /** Specifies the merge engine for table with primary key. */
     public enum MergeEngine implements DescribedEnum {
-        DEDUPLICATE("deduplicate", "De-duplicate and keep the last row."),
+        DEDUPLICATE("deduplicate", "De-duplicate and keep the last row.", true, true),
 
-        PARTIAL_UPDATE("partial-update", "Partial update non-null fields."),
+        PARTIAL_UPDATE("partial-update", "Partial update non-null fields.", false, false),
```

Comment:
When using partial-update and no col contains agg, update is theoretically supported